sos/plugins/gnocchi.py

```diff
@@ -70,18 +56,64 @@ def setup(self):
                 "gnocchi resource-type list"
             ])
 
-    def postproc(self):
-        self.do_file_sub(
-            "/etc/gnocchi/gnocchi.conf",
-            r"password=(.*)",
-            r"password=*****",
+    def apply_regex_sub(self, regexp, subst):
+        self.do_path_regex_sub("/etc/gnocchi/*", regexp, subst)
+        self.do_path_regex_sub(
+            self.var_puppet_gen + "/etc/gnocchi/*",
+            regexp, subst
         )
-        self.do_file_sub(
-            self.var_puppet_gen + "/etc/gnocchi/"
-            "gnocchi.conf",
-            r"password=(.*)",
-            r"password=*****",
+
+    def postproc(self):
+        protect_keys = ["ceph_secret", "password", ]
+        self.apply_regex_sub(
+            r"((?m)^\s*(%s)\s*=\s*)(.*)" % "|".join(protect_keys),
+            r"\1*********"
         )
 
 
+class RedHatGnocchi(Gnocchi, RedHatPlugin):
+
+    packages = (
+        'openstack-gnocchi-metricd', 'openstack-gnocchi-common',
+        'openstack-gnocchi-statsd', 'openstack-gnocchi-api',
+        'openstack-gnocchi-carbonara'
+    )
+
+    def setup(self):
+        super(RedHatGnocchi, self).setup()
+        self.add_copy_spec([
+            self.var_puppet_gen + "/etc/gnocchi/*",
+            self.var_puppet_gen + "/etc/httpd/conf/*",
+            self.var_puppet_gen + "/etc/httpd/conf.d/*",
+            self.var_puppet_gen + "/etc/httpd/conf.modules.d/wsgi.conf",
+            self.var_puppet_gen + "/etc/my.cnf.d/tripleo.cnf"
+        ])
+
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/httpd/gnocchi*",
+            ])
+        else:
+            self.add_copy_spec([
+                "/var/log/httpd/gnocchi*.log",
+            ])
```

Comment:
`add_copy_spec()` supports both lists and strings for single file specifications. No need for a list here.